src/masternodes/mn_checks.cpp

```diff
@@ -2979,17 +2979,25 @@ class CCustomTxApplyVisitor : public CCustomTxVisitor
         }
 
         // Calculate DFI and DUSD value separately
-        uint64_t totalCollateralsDUSD = 0;
-        uint64_t totalCollateralsDFI = 0;
+        CAmount totalCollateralsDUSD = 0;
+        CAmount totalCollateralsDFI = 0;
+        CAmount factorDUSD = 0;
+        CAmount factorDFI = 0;
 
         for (auto& col : collateralsLoans.collaterals){
-            if (col.nTokenId == DCT_ID{0} )
+            auto token = mnview.GetCollateralTokenFromAttributes(col.nTokenId);
+
+            if (col.nTokenId == DCT_ID{0} ){
                 totalCollateralsDFI += col.nValue;
+                factorDFI = token->factor;
+            }
 
-            if(tokenDUSD && col.nTokenId == tokenDUSD->first)
+            if(tokenDUSD && col.nTokenId == tokenDUSD->first){
                 totalCollateralsDUSD += col.nValue;
+                factorDUSD = token->factor;
+            }
         }
-        auto totalCollaterals = totalCollateralsDUSD + totalCollateralsDFI;
+        auto totalCollaterals = MultiplyAmounts(totalCollateralsDUSD, factorDUSD) + MultiplyAmounts(totalCollateralsDFI, factorDFI);
```

Comment:
Will need a fork condition unfortunately. Let's do the following:

- Separate into two different lambas inside CollateralPctCheck. 
  - PreGreatWorldCollaterals, 
  - PostGreatWorldCollaterals
  
 and abstract this logic out into these functions.

Reply:
You mean pre/postGrandCentral or am I missing something?

Reply:
if this is released without a hardfork/blockheight-gate, you risk chain split if a takeLoan or withdraw happens that is allowed in the updated version, but not in the old version. So you need to ensure that all active nodes have the updated version before it becomes active -> fork condition

Reply:
Yes I understand this however, this was introduced in Fort Canning Epilogue (v2.11.1) and I guess the targeted version for the changes is the next fork Grand Central(v3) and not Fort Canning Great World (v2.10.0). So the conditions should be pre/postGrandCentral unless we are planning an intermediate fork which I am not aware of.

Reply:
sorry yes. my bad, didn't read properly. agreed that this will be fine in GrandCentral.